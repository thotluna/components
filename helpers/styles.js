import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
