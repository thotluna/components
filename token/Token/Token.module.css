.container {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
}

.description {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-left: 10px;
  gap: 4px;
}

.label {
  min-width: max-content;
  padding: 4px 8px;
  border: 1px solid var(--button-base-color-border);
  background-color: var(--color-base-transparent);
  border-radius: var(--border-radius-sm);
  color: var(--color-font-base);
  font-family: var(--font-family-mono);
  font-size: var(--font-size-sm);
}

.label:first-child {
  border: var(--border-width-thin) solid var(--button-base-color-border);
  background-color: var(--color-base-transparent);
}

.color {
  width: 65px;
  height: 65px;
  border: var(--button-base-color-border);
  border-radius: var(--border-radius-full);
  box-shadow: var(--box-shadow-sm);
}

.spacer {
  display: flex;
  width: 50px;
  height: 50px;
  align-items: center;
  justify-content: center;
}

.font-size,
.font-family {
  width: 100%;
  max-width: 500px;
  text-align: justify;
}

.font-size {
  text-align: right;
}

.title {
  font-family: var(--font-family-title);
  font-size: 16px;
}

.sans {
  font-family: var(--font-family-sans);
  font-size: 16px;
}

.box {
  width: 20rem;
  height: 5rem;
  border: 1px solid black;
}
