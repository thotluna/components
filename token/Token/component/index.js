export { default as Color } from './Colors/Colors'
export { default as Spacing } from './Spacing/Spacing'
export { default as FontFamily } from './Fonts/FontFamily'
export { default as FontSize } from './Fonts/FontSize'
export { default as FontWeight } from './Fonts/FontWeight'
