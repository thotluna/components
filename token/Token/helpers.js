import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
