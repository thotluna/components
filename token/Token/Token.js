import React from 'react'

import { Color, Spacing, FontFamily, FontSize, FontWeight } from './component'

const Token = () => <></>

Token.Color = Color
Token.Spacing = Spacing
Token.FontFamily = FontFamily
Token.FontSize = FontSize
Token.FontWeight = FontWeight

export default Token
