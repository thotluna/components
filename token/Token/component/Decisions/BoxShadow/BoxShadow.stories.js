import Token from '../../..'
import { choices } from '../../../..'
import { decisions } from '../../../..'

import { getTemplate, getListTemplate } from '../../../../../helpers/storybook'
import { getTokenItems } from '../../../helpers'

export default {
  title: 'Tokens/Box-Shadow',
  component: Token.BoxShadow,
  args: {},
  argTypes: {
    value: {
      control: {
        type: 'select',
        options: Object.keys(choices.boxShadow).map(
          (key) => choices.boxShadow[key]
        ),
      },
    },
    isPlayground: { table: { disable: true } },
    label: { table: { disable: true } },
  },
}

const Template = getTemplate(Token.BoxShadow, {
  fontFamily: 'Open Sans',
  fontSize: '1.6rem',
})

export const Default = Template.bind({})
Default.args = {
  label: '--box-shadow-none',
  value: '0',
}

const getItems = getTokenItems(choices.boxShadow, 'boxShadow')
const ListTemplate = getListTemplate(Token.BoxShadow)

export const Familys = ListTemplate.bind({})
Familys.args = {
  items: getItems(),
}

const getDecisionLightItem = getTokenItems(decisions.color.light, 'color')

export const Light = ListTemplate.bind({})
Light.args = {
  __sb: { fd: 'row' },
  items: getDecisionLightItem('boxShadow'),
}

const getDecisionDarkItem = getTokenItems(decisions.color.dark, 'color')

export const Dark = ListTemplate.bind({})
Dark.args = {
  __sb: { fd: 'row' },
  items: getDecisionDarkItem('boxShadow'),
}
