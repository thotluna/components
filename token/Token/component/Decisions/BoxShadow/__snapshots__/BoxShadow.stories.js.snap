// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Tokens/Box-Shadow Dark 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "none",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-boxShadow-none
      </span>
      <span
        className="label"
      >
        none
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "0px 3px 6px rgba(255, 255, 255, 0.05)",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-boxShadow-xs
      </span>
      <span
        className="label"
      >
        0px 3px 6px rgba(255, 255, 255, 0.05)
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "0px 4px 5px rgba(255, 255, 225, 0.07)",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-boxShadow-sm
      </span>
      <span
        className="label"
      >
        0px 4px 5px rgba(255, 255, 225, 0.07)
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Box-Shadow Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "0",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --box-shadow-none
      </span>
      <span
        className="label"
      >
        0
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Box-Shadow Familys 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "none",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --box-shadow-none-t
      </span>
      <span
        className="label"
      >
        none
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "0px 3px 6px rgba(0, 0, 0, 0.05)",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --box-shadow-xs-t
      </span>
      <span
        className="label"
      >
        0px 3px 6px rgba(0, 0, 0, 0.05)
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "0px 4px 5px rgba(0, 0, 0, 0.07)",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --box-shadow-sm-t
      </span>
      <span
        className="label"
      >
        0px 4px 5px rgba(0, 0, 0, 0.07)
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "0 0 0 1px var(--color-primary)",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --box-shadow-focus-in-t
      </span>
      <span
        className="label"
      >
        0 0 0 1px var(--color-primary)
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "0 0 10px 0 var(--color-primary)",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --box-shadow-focus-out-t
      </span>
      <span
        className="label"
      >
        0 0 10px 0 var(--color-primary)
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Box-Shadow Light 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "none",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-boxShadow-none
      </span>
      <span
        className="label"
      >
        none
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "0px 3px 6px rgba(0, 0, 0, 0.05)",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-boxShadow-xs
      </span>
      <span
        className="label"
      >
        0px 3px 6px rgba(0, 0, 0, 0.05)
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "boxShadow": "0px 4px 5px rgba(0, 0, 0, 0.07)",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-boxShadow-sm
      </span>
      <span
        className="label"
      >
        0px 4px 5px rgba(0, 0, 0, 0.07)
      </span>
    </div>
  </div>
</div>
`;
